### DISCUSSION TITLE
Filter search results on existence of a file in a zip

### DISCUSSION THREAD

**Discussion by juxeii** [junegunn/fzf#3184] [Q&A]
Hi,

how would you do the following:
1) Filter for `*.zip` files in folder `bar`
2) Look inside each zip file and look for a file called `foo.cpp`
3) If this file is present, keep the zip file in the result list for fzf

To check for the existence of `foo.cpp` I could go with something like 
`unzip -l foo.cpp`
and check if this returns something.
What is the simplest approach you can think of for achieving this?
I do not know how to include arbitrary predicates in fzf.
I would like to work with ripgrep.

**Comment by sweetbbak** [comment]
`cd bar && fd --extension zip --exec unzip -l {} \; | rg '.*cpp' | fzf --cycle --reverse`

I'm not exactly sure what it is that you are trying to do but this snippet will find all files/folders ending with .zip, it will list the contents of the zip file and ripgrep will search for files that end with .cpp and then pipe them into fzf so that you can see and select them. From there you could pipe into into another command. You can also replace fd with the standard find command.

the "\;" is necessary for running commands that fd finds. You could make this a lot more complex, Id look at the ripgrep/fzf script that is in the readme, its a good example of how you can achieve something like this. there is probably a better way to view the contents of a zip folder but I dont know it. lzcat and lz are potentially good things to look into

**Reply by juxeii** [reply]
Thanks, really helpful.

**Reply by sweetbbak** [reply]
No problem, glad it helped.
